pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -69,32 +69,31 @@ func (o *objectWatcherImpl) Create(clusterName string, desireObj *unstructured.U
 	// users should resolve the conflict in person.
 	clusterObj, err := dynamicClusterClient.DynamicClientSet.Resource(gvr).Namespace(desireObj.GetNamespace()).Create(context.TODO(), desireObj, metav1.CreateOptions{})
 	if err != nil {
+		existObj, err := dynamicClusterClient.DynamicClientSet.Resource(gvr).Namespace(desireObj.GetNamespace()).Get(context.TODO(), desireObj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get exist resource(kind=%s, %s/%s) in cluster %v: %v", desireObj.GetKind(), desireObj.GetNamespace(), desireObj.GetName(), clusterName, err)
+		}
 		// The 'IsAlreadyExists' conflict may happen in following known scenarios:
 		// - 1. In a reconcile process, the execution controller successfully applied resource to member cluster but failed to update the work conditions(Applied=True),
 		//   when reconcile again, the controller will try to apply(by create) the resource again.
 		// - 2. The resource already exist in the member cluster but it's not created by karmada.
 		if apierrors.IsAlreadyExists(err) {
-			existObj, err := dynamicClusterClient.DynamicClientSet.Resource(gvr).Namespace(desireObj.GetNamespace()).Get(context.TODO(), desireObj.GetName(), metav1.GetOptions{})
-			if err != nil {
-				return fmt.Errorf("failed to get exist resource(kind=%s, %s/%s) in cluster %v: %v", desireObj.GetKind(), desireObj.GetNamespace(), desireObj.GetName(), clusterName, err)
-			}
-
 			// If the existing resource is managed by Karmada, then just update it.
 			if util.GetLabelValue(desireObj.GetLabels(), workv1alpha1.WorkNameLabel) == util.GetLabelValue(existObj.GetLabels(), workv1alpha1.WorkNameLabel) {
 				return o.Update(clusterName, desireObj, existObj)
```

Comment:
`apierrors.IsAlreadyExists(err)`  will it be false forever?

when will this code run?

Reply:
> `apierrors.IsAlreadyExists(err)` will it be false forever?
> 
> when will this code run?

if promote a nodeport service from member cluster to control plane,`apierrors.IsAlreadyExists(err)` is true